### PR TITLE
cleanup: move gRPC centric code generation to a gRPC specific file

### PR DESCRIPTION
**Pull request by software-dov**
This is a followup to an earlier review where it was stated that
moving gRPC centric code to its own file was a worthwhile maintenance
task.

The diff in this commit is large but is almost entirely moving around
existing code blocks with no structural or behavioral changes.
Any new code is just boilerplate intended to check which transports
the generator should emit code for and dispatch accordingly, or
renaming existing methods to indicate their gRPC centric nature.

### CODE REVIEW

File: internal/gengapic/gengrpc.go [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
```
Comment (codyoss) [inline internal/gengapic/gengrpc.go:21]: Could migrate to `google.golang.org/protobuf/proto`. But that could/should be its own PR.

**Review by codyoss — COMMENTED** [review]
LGTM, but I'll defer to your team that knows this code better.

File: internal/gengapic/gengrpc.go (hunk 1 above)
Comment (noahdietz) [reply internal/gengapic/gengrpc.go:21]: We should definitely start using the v2 module for all helper methods where possible, but migrating the plugin implementation to v2 will be a tall task so we may not be able to do everything. 

+1 to a separate PR though

Comment (software-dov) [reply internal/gengapic/gengrpc.go:21]: Agreed, that belongs in another PR.

**Review by tbpg — APPROVED** [review]
